File: FitnessApp/app/src/main/java/com/example/fitnessapp/data/model/CreateCompletionRequest.kt
package com.example.fitnessapp.data.model

data class CompletionExerciseItem(
    val exerciseId: Int,
    val weight: Float
)

data class CreateCompletionRequest(
    val completedAt: String? = null,
    val exercises: List<CompletionExerciseItem>
)

File: FitnessApi/FitnessApi/Models/CreateCompletionRequest.cs
namespace FitnessApi.Models
{
    public class CreateCompletionRequest
    {
        public DateTime? CompletedAt { get; set; }
        public List<ExerciseCompletionDto> Exercises { get; set; }
    }

    public class ExerciseCompletionDto
    {
        public int ExerciseId { get; set; }
        public float Weight { get; set; }
    }
}
